Source/DafnyCore/AST/IHasUsages.cs

```diff
@@ -10,8 +11,33 @@ public interface IHasUsages : IDeclarationOrUsage {
   public IEnumerable<IDeclarationOrUsage> GetResolvedDeclarations();
 }
 
+public static class AstExtensions {
+
+  public static string GetMemberQualification(MemberDecl memberDecl) {
+    return memberDecl.EnclosingClass.Name == "_default" ? "" : $"{memberDecl.EnclosingClass.Name}.";
+  }
+
+  /// <summary>
+  /// Returns a text representation of the given variable.
+  /// </summary>
+  /// <param name="variable">The variable to get a text representation of.</param>
+  /// <returns>The text representation of the variable.</returns>
+  public static string AsText(this IVariable variable) {
+    var ghost = variable.IsGhost ? "ghost " : "";
+    string type;
+    try {
+      type = variable.Type.ToString();
+    } catch (Exception e) {
+      type = $"<Internal error: {e.Message}>";
+    }
+    return $"{ghost}{variable.Name}: {type}";
+  }
+}
+
 public interface ISymbol : IDeclarationOrUsage {
   DafnySymbolKind Kind { get; }
+
+  string GetHoverText(DafnyOptions options);
```

Comment:
Since this is in the core file and not in the LSP, would it make sense to rename it GetSymbolInfo? There is no hover in the command line.